cpp/src/jit/parser.hpp

```diff
@@ -213,17 +209,16 @@ class ptx_parser {
  * @param src The input PTX code.
  * @param function_name The User defined function that the output CUDA function
  * will have.
- * @param output_arg_type output_arg_type The C++ type of the output parameter of the
- * function
- * @param pointer_arg_list A list of the parameters that are pointers.
+ * @param param_types the types of the parameters of the function. each entry should map the
+ * parameter index to the CUDA type of the parameter. The unspecified indices will be have types
+ * based on the PTX's parameter types.
  * @return The output CUDA device function
  */
 inline std::string parse_single_function_ptx(std::string const& src,
```

Comment:
Could this implementation be move to the .cpp file? I don't see a value in making it inline.

Reply:
yeah, sure!